src/test/java/uk/gov/companieshouse/overseasentitiesapi/model/DtoModelChangeTest.java

```diff
@@ -15,37 +16,75 @@
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.fail;
 import static uk.gov.companieshouse.overseasentitiesapi.configuration.MongoConverters.getMongoCustomConversions;
 
 /**
  * If any of the tests in this class start failing it's likely due to a DTO model structure change which is an indication
- * of potential issues for clients that still need to be able to work with Mongo records created using older versions of
- * the data model.
+ * of potential - current or future - issues for clients that still need to be able to work with Mongo records created
+ * using older versions of the data model.
  *
- * If the data model structure has been modified, consideration will need to be given to how the web - and other -
+ * If the data model structure has been modified, consideration needs to be given to how the web - and other -
  * clients of the OE API end-points will deal with older versions of the model structure and ultimately how older
- * versions of the model structure can be loaded into the newly changed DAO and DTO models. Changes will probably
- * need to be made to the Mongo custom 'converters'.
+ * versions of the model structure can be loaded into the newly changed DAO and DTO models. Among other things, a new,
+ * Mongo, custom 'transformer' class may need to be written.
  *
- * Once those things have been considered and addressed, existing tests should pass again. Then, a file with the new JSON
- * model structure should be added to the test resources folder and a new test added to successfully read and process
- * the new, up-to-date DAO/DTO structure.
+ * If a non-breaking change to the model has been made - e.g. a new data field has simply been added to a DTO and DAO -
+ * the 'current' test JSON file needs updating to avoid a null check failure (this essentially future-proofs the test,
+ * which will potentially fail if the datatype of the newly added field ever changes).
  *
- * Note the use of Spring profiles, to ensure that the (default) <code>MongoConfig</code> configuration is not loaded
- * during this test (as the specific in-memory Mongo DB config needs to be picked up).
+ * If a breaking change to the model has been made - e.g. a data field has changed its type from String to a custom POJO -
+ * after a new 'transformer' has been implemented, a file with the new JSON model structure should be added to the test
+ * resources folder and the <code>OLD_JSON_MODEL_FILENAMES</code> and <code>CURRENT_JSON_MODEL_FILENAME</code> constants
+ * updated, in order that the new, up-to-date DAO/DTO structure is fully tested.
+ *
+ * Finally, note the use of Spring profiles, to ensure that the (default) <code>MongoConfig</code> configuration is not
+ * loaded during this test (as the specific in-memory Mongo DB config needs to be picked up).
  */
 @AutoConfigureDataMongo
 @SpringBootTest
 @ActiveProfiles("test")
```

Comment:
i know it's not new but this test class seems to be the only one with @ActiveProfiles("test"), perhaps remove it so it always runs with the other tests?

Reply:
It's (still) needed - see comment on line 59. If not present, test fails to initialise and run.